utils/utils.py

```diff
@@ -61,6 +76,10 @@ def setup_ddp():
     return distributed_data_parallelism, world_size, world_rank
 
 
+def kill_mpi_environment():
+    MPI.Finalize()
```

Comment:
Not sure the right way to ensure other processes continue in the session and only the last one calls